.editorconfig

```diff
@@ -0,0 +1,7 @@
+root = true
+
+[*]
+indent_size = 4
+indent_style = tab
+insert_final_newline = true
+trim_trailing_whitespace = true
```

Comment:
what do you think about moving this to a separate pull request?
I think it should be a little bit more sophisticated:

* limit this style to [*.{cpp,hpp,c,h}]
* don't specify an ``indent_size``
* add style for [*.py] to use spaces and match the indent size that's used there
* maybe add a style to .travis and appveyor.yml as well